src/jlrichdem/depressions/jldepressions.cpp

```diff
@@ -0,0 +1,30 @@
+#include <jlcxx/jlcxx.hpp>
+#include <jlcxx/stl.hpp>
+#include <typeinfo>
+
+#include <richdem/common/Array2D.hpp>
+#include <richdem/common/constants.hpp>
+#include <richdem/depressions/depression_hierarchy.hpp>
+#include <richdem/depressions/fill_spill_merge.hpp>
+namespace rd = richdem;
+
+JLCXX_MODULE define_depressions_module(jlcxx::Module &mod)
+{
+    jlcxx::TypeWrapper<rd::dephier::Depression<double>> depression_double =
+        mod.add_type<rd::dephier::Depression<double>>("DepressionDouble");
+
+    jlcxx::TypeWrapper<rd::dephier::DepressionHierarchy<double>> depression_hierarchy_double =
+        mod.add_type<rd::dephier::DepressionHierarchy<double>>("DepressionHierarchyDouble");
```

Comment:
Would it be possible for the julia type to also be parametric, i.e. DepressionHierarchy{Float64}?

Reply:
In this case I just took the shortcut and created `DepressionHierarchyDouble` and not a pure parametric type like `DepressionHierarchy{Float64}` since it is not being used in any other sense. Do you think you will be needing it in any other sense ?